assets/src/components/vehicleIcon.tsx

```diff
@@ -1,16 +1,299 @@
 import React from "react"
 
-const VehicleIcon = ({ scale }: { scale?: number }) => {
-  const scaleAmount = scale || 1.0
+export enum Orientation {
+  Up,
+  Down,
+  Left,
+  Right,
+}
+
+export enum Size {
+  Small,
+  Medium,
+  Large,
+}
+
+export interface Props {
+  size: Size
+  orientation: Orientation
+  label?: string
+  variant?: string | null
```

Comment:
Doesn't the `?` mean you don't need the `| null`?

Reply:
The `?` is just for `undefined`. We can get either an `undefined` if you don't pass a `<VehicleIcon variant={...}` property or a `null` if you do `<VehicleIcon variant={vehicle.via_variant}` but the `via_variant` is `null`.

Reply:
I see, thanks. I guess the potential problem is via_variant being null instead of undefined. (But not a problem to be addressed right now.)